### PR TITLE
WEB-2309: Initial implementation

### PR DESCRIPTION
**Pull request by nitzanashi**
examples: https://github.com/zattoo/web-cactus-test/actions/workflows/milestones.yml

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,116 +1,139 @@
...
 
-## Change action.yml
 
-The action.yml defines the inputs and output for your action.
+| Parameter     | Type      |      Description      |  Required  |
+|---------------|:---------:|---------------------|:----------:|
+| `token`       | `string`  | GitHub token          | `true`     |
+| `milestone`   | `string`  | Milestone title       | `true`     |
+| `description` | `string`  | Description of the milestone | `false` |
+| `deu_on`      | `date as string` | The date which the milestone shall be due on | `false` |
```
Comment (mxkyb) [inline README.md:46]:
```suggestion
| `due_on`      | `date as string` | The date which the milestone shall be due on | `false` |
```

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            }
+
+            const issue = core.getInput('issue', {required: false});
+
+            if (!issue) {
+                throw new Error(`Can't assign milestone to pull-request, issue parameter is missing`);
+            }
+
+            // see https://docs.github.com/en/github/searching-for-information-on-github/searching-on-github/searching-issues-and-pull-requests
+            const q = `${issue} is:pr in:title repo:${owner}/${repo}`;
```
Comment (mxkyb) [inline src/index.js:119]: I think the naming `q` is bad. maybe `query`, if thats what you mean?

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            break;
+        }
+
+        default: {
+            throw new Error(`Unknown event no handler found for ${context.eventName}\nsupported events are ${Object.values(events).join()}`);
+        }
+    }
+})().catch((error) => {
+  core.setFailed(error);
+  process.exit(1);
```
Comment (mxkyb) [inline src/index.js:153]: why exit with explicit error code after `core.setFailed`?

File: src/index.js (hunk 2 above)
Comment (nitzanashi) [reply src/index.js:119]: I tend to agree with you but the api [api](https://docs.github.com/en/rest/reference/search#search-issues-and-pull-requests) require `q` as parameter, so I tried to safe some code

**Comment by gotbahn** [comment]
where I can see example? one in header running different action

**Comment by nitzanashi** [comment]
> where I can see examples? one in description running different action

Nono it's the same action

**Comment by gotbahn** [comment]
hm... let's try to catch up tomorrow to explain me how it works

File: README.md [hunk 4]
```diff
@@ -1,116 +1,139 @@
...
+```yml
+name: Milestones
+
+on:
+  repository_dispatch:
+    types:
+      - create_milestone
+      - update_milestone
+      - assign_milestone
+      - close_milestone
```
Comment (gotbahn) [inline README.md:19]:
```suggestion
      - milestone_create
      - milestone_update
      - milestone_assign
      - milestone_close
```

**Review by gotbahn — APPROVED** [review]

**Pull request merged**